File: JavaScript projects/JavaScript Basics/Conditional Statements Advanced/Exercises/03_new_house.js
function newHouse(flower, quantity, budget){
    let price;
    let discount = 1; 
    if (flower == "Roses"){
        price = 5.0;
        if (80 < quantity){
            discount = 0.9;
        }
    }else if (flower == "Dahlias"){
        price = 3.8;
        if (90 < quantity){
            discount = 0.85;
        }
    }else if (flower == "Tulips"){
        price = 2.8;
        if (80 < quantity){
            discount = 0.85;
        }
    }else if (flower == "Narcissus"){
        price = 3.0;
        if (quantity < 120){
            discount = 1.15;
        }
    }else if (flower == "Gladiolus"){
        price = 2.5;
        if (quantity < 80){
            discount = 1.2;
        }
    }
    let totalPrice = quantity * price * discount;
    if (totalPrice <= budget){
        console.log(`Hey, you have a great garden with ${quantity} ${flower} and ${(budget - totalPrice).toFixed(2)} leva left.`);
    }else{
        console.log(`Not enough money, you need ${(totalPrice - budget).toFixed(2)} leva more.`);
    }
}
newHouse("Tulips",88,260);

File: Python projects/Python Basics/03_conditional_statements_advanced/exercises/03_new_house.py
flower_type = str(input())
quantity_flowers = int(input())
budget = int(input())
flower_cost = 0.0  # the else-conditions regarding the price that will not be affected by discount, could be assigned
total_cost = 0.0   # to total_cost before the if- condition that checks the amount, would save 1 line per each type of
# flower, furthermore prices could be set before the conditions

if flower_type == "Roses":
    flower_cost = 5
    if quantity_flowers > 80:
        total_cost = quantity_flowers * flower_cost * 0.9
    else:
        total_cost = quantity_flowers * flower_cost
elif flower_type == "Dahlias":
    flower_cost = 3.8
    if quantity_flowers > 90:
        total_cost = quantity_flowers * flower_cost * 0.85
    else:
        total_cost = quantity_flowers * flower_cost
elif flower_type == "Tulips":
    flower_cost = 2.8
    if quantity_flowers > 80:
        total_cost = quantity_flowers * flower_cost * 0.85
    else:
        total_cost = quantity_flowers * flower_cost
elif flower_type == "Narcissus":
    flower_cost = 3
    if quantity_flowers < 120:
        total_cost = quantity_flowers * flower_cost * 1.15
    else:
        total_cost = quantity_flowers * flower_cost
else:
    flower_cost = 2.5
    if quantity_flowers < 80:
        total_cost = quantity_flowers * flower_cost * 1.2
    else:
        total_cost = quantity_flowers * flower_cost

if budget >= total_cost:
    print(
        f"Hey, you have a great garden with {quantity_flowers} {flower_type} and {budget - total_cost:.2f} leva left.")
else:
    print(f"Not enough money, you need {total_cost - budget:.2f} leva more.")
